js/packages/common/src/contexts/meta/pullAuctions.ts

```diff
@@ -0,0 +1,95 @@
+import { Connection } from '@solana/web3.js';
+
+import { getMultipleAccounts } from '../accounts/getMultipleAccounts';
+import { AuctionManagerV1, AuctionManagerV2 } from '../../models';
+import { ParsedAccount } from '../accounts/types';
+import { AUCTION_ID } from '../../utils/ids';
+import { BIDDER_METADATA_LEN, BIDDER_POT_LEN } from '../../actions';
+import { getProgramAccounts } from './web3';
+import { AccountAndPubkey } from './types';
+
+type AuctionManagerByAuction = Record<
+  string,
+  ParsedAccount<AuctionManagerV1 | AuctionManagerV2>
+>;
+
+interface IPullAuctionsParams {
+  connection: Connection;
+  auctionManagersByAuction: AuctionManagerByAuction;
+}
+
+export const pullAuctions = async ({
+  connection,
+  auctionManagersByAuction,
+}: IPullAuctionsParams) => {
+  const IDS_PER_CHUNK = 100;
+  const auctionIdsChunks = [
+    ...chunks(
+      getAuctionsIdsByAuctionManager(auctionManagersByAuction),
+      IDS_PER_CHUNK,
+    ),
+  ];
+
+  const auctionsChunks = auctionIdsChunks.map(ids =>
+    pullAuctionsByAuctionManager({ ids, connection }),
+  );
+
+  const bidderMetadata = getProgramAccounts(connection, AUCTION_ID, {
+    filters: [
+      // Filter for BidderMetadata by data size
+      {
+        dataSize: BIDDER_METADATA_LEN,
+      },
+    ],
+  });
+
+  const bidderPots = getProgramAccounts(connection, AUCTION_ID, {
+    filters: [
+      // Filter for BidderPot by data size
+      {
+        dataSize: BIDDER_POT_LEN,
+      },
+    ],
+  });
```

Comment:
This is better than querying all bidder pots and metadata by the auction manager in the init page load. However, as the program grows this will eventually cause processing delays on the frontend as it parses and tosses out accounts not needed for the storefront.